File: 2025_08/leetcode2025-08-16.py
class Solution:
    def maximum69Number(self, num: int) -> int:
        s = str(num)
        for i in range(len(s)):
            if s[i] == '6':
                s = s[:i]+'9'+s[i+1:]
                return int(s)
        return num

if __name__ == "__main__":
    s = Solution()
    num = 9669
    print(s.maximum69Number(num))

File: leetcode2025/2025_08/leetcode2025-08-16.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution {
public:
    int maximum69Number (int num) {
        string s = to_string(num);
        for (int i=0; i<s.size(); i++){
            if (s[i] == '6'){
                s[i] = '9';
                return stoi(s);
            }
        }
        return num;
    }
};

int main(){
    Solution s;
    int num = 9669;
    cout << s.maximum69Number(num) << endl;
}
